docs/dev/data_validation.rst

```diff
@@ -0,0 +1,688 @@
+These packages include a bunch of useful tests that can be applied to any table.
+There are several examples of applying tests from ``dbt-expectations`` in
+``dbt/models/vcerare/out_vcerare__hourly_available_capacity_factor/schema.yml``
+and in general they will look like the below. Each item in a ``data_tests`` section
+defines a single test, and may provide named parameters for the test. The tests whose
+names have the ``dbt_expectations`` prefix come from that package.
+
+.. code-block:: yaml
+
+    version: 2
+    sources:
+      - name: pudl
+        tables:
+          - name: out_vcerare__hourly_available_capacity_factor
+            data_tests:
+              - expect_columns_not_all_null
+              - check_row_counts_per_partition:
+                  table_name: out_vcerare__hourly_available_capacity_factor
+                  partition_column: report_year
+              - expect_valid_hour_of_year
+              - expect_unique_column_combination:
+                  columns:
+                    - county_id_fips
+                    - datetime_utc
+            columns:
+              - name: state
+                data_tests:
+                  - not_null
+              - name: place_name
+                data_tests:
+                  - not_null
+                  - dbt_expectations.expect_column_values_to_not_be_in_set:
+                      value_set:
+                        - bedford_city
+                        - clifton_forge_city
+                        - lake_hurron
+                        - lake_st_clair
+                  - dbt_expectations.expect_column_values_to_be_in_set:
+                      value_set:
+                        - oglala lakota
+                      row_condition: "county_id_fips = '46012'"
+              - name: datetime_utc
+                data_tests:
+                  - not_null
+                  - dbt_expectations.expect_column_values_to_not_be_in_set:
+                      value_set:
+                        - "{{ dbt_date.date(2020, 12, 31) }}"
+              - name: report_year
+                data_tests:
+                  - not_null
+              - name: hour_of_year
+                data_tests:
+                  - not_null
+                  - dbt_expectations.expect_column_max_to_be_between:
+                      min_value: 8760
+                      max_value: 8760
+
+
+Tests defined within PUDL
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some of the tests in the example above like ``expect_columns_not_all_null`` or
+``check_row_counts_per_partition`` are defined by us, and can be found in the SQL
+files with the same name under ``dbt/tests/data_tests/generic_tests/``
+
+Documentation for the tests that we define is in
+``dbt/testse/data_tests/generic_tests/schema.yml``
+
+.. todo::
+
+   * Integrate documentation of our existing generic tests into the docs build.
+   * Convert all bespoke / singular tests into generic tests.
```

Comment:
@krivard I haven't looked into converting these yet. Do you think it's easy enough that we should just do it in this PR so that we neither need to document them, or leave them undocumented?

Reply:
You did it! :tada: